presto-pinot-toolkit/src/main/java/com/facebook/presto/pinot/query/PinotAggregationProjectConverter.java

```diff
@@ -217,17 +231,45 @@ private PinotExpression handleFunction(
             CallExpression function,
             Map<VariableReferenceExpression, PinotQueryGeneratorContext.Selection> context)
     {
-        switch (function.getDisplayName().toLowerCase(ENGLISH)) {
+        String functionName = function.getDisplayName().toLowerCase(ENGLISH);
+        switch (functionName) {
             case "date_trunc":
                 boolean useDateTruncation = PinotSessionProperties.isUseDateTruncation(session);
                 return useDateTruncation ?
                         handleDateTruncationViaDateTruncation(function, context) :
                         handleDateTruncationViaDateTimeConvert(function, context);
+            case "array_max":
+            case "array_min":
+                return derived(String.format("%s(%s)",
+                        ARRAY_AGGREGATION_ALLOWED_PUSHDOWN_MAP.get(functionName),
+                        function.getArguments().get(0).accept(this, context).getDefinition()));
+            // array_sum and array_reduce are translated to a reduce function with lambda functions, so we pass in
+            // this arrayVariableHint to help determine which array function it is.
+            case "reduce":
+                if (arrayVariableHint != null) {
+                    String arrayFunctionName = getArrayFunctionName(arrayVariableHint);
+                    if (arrayFunctionName != null) {
+                        String inputColumn = function.getArguments().get(0).accept(this, context).getDefinition();
+                        return derived(String.format("%s(%s)", arrayFunctionName, inputColumn));
+                    }
+                }
             default:
                 throw new PinotException(PINOT_UNSUPPORTED_EXPRESSION, Optional.empty(), format("function %s not supported yet", function.getDisplayName()));
         }
     }
 
+    // The array function variable names are in the format of `array_sum`, `array_average_0`, `array_sum_1`.
+    // So we can parse the array function name based on variable name.
+    private String getArrayFunctionName(VariableReferenceExpression variable)
+    {
+        String[] variableNameSplits = variable.getName().split("_");
+        if (variableNameSplits.length < 2) {
+            return null;
+        }
+        String arrayFunctionName = String.format("%s_%s", variableNameSplits[0], variableNameSplits[1]);
```

Comment:
I feel we should do a stronger check here: Like that the split length should not be more than 3 perhaps ? Just vary of doing the wrong thing if say we get something weird like array_sum_1_3 :-P

Reply:
done